pages/rating.vue

```diff
@@ -0,0 +1,66 @@
+<script setup lang="ts">
+import type { Game, GameStats, User } from "@prisma/client";
+
+type GameWithStatsAndUser = Game & {
+  game_stats: (GameStats & {
+    user: User;
+  })[];
+};
+
+type gameStats = {
+  game_id: number;
+  start_time: string;
+  end_time: string;
+  end_reason: string;
+  game_stats: Array<unknown>;
+};
+
+const games = ref<GameWithStatsAndUser[]>([]);
+const isLoading = ref(true);
+
+onMounted(async () => {
+  console.log("Fetching games...");
+
+  try {
+    const response = await fetch("/api/rating");
+    const data = await response.json();
+
+    if (Array.isArray(data)) {
+      games.value = data.map((game: gameStats) => ({
+        ...game,
+        start_time: new Date(game.start_time),
+        end_time: new Date(game.end_time),
+      })) as GameWithStatsAndUser[];
+      isLoading.value = false;
+    } else {
+      console.warn("Unexpected data format:", data);
+    }
+  } catch (error) {
+    console.error("Failed to fetch games:", error);
+  }
+});
```

Comment:
Если использовать `useFetch`, то:
- не нужен `try...catch`
- `rating` будет автоматически правильно типизирован в соответствии с тем типом, который ты возвращаешь в `rating.get.ts`
- `isLoading` можно либо убрать вообще и использовать вместо него `status`, либо рассчитать из `status` (не нужен `isLoading` ref)

`ref` переменную `games` я бы тоже убрал для простоты и расчитывал бы её вот так, как показал ниже. Типы для неё тоже скорее всего, можно удалить, т.к. TypeScript автоматически поймёт типы исходя из типов `rating` и того, что ты делаешь с переменной.

В идеале, все преобразования над рейтингом стоит делать на бакенде, а на фронтенде просто отображать его.

Кстати, а разве `start_time` и `end_time` у нас уже не `Date`?

```suggestion
const { data: rating, status } = await useFetch("/api/rating");
const isLoading = status === "pending";

const games = rating.map(game => ({
  ...game,
  start_time: new Date(game.start_time),
  end_time: new Date(game.end_time),
}));
```